spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -447,7 +447,7 @@ private ResultSet executeRead(String tableName, KeySet keys, Iterable<String> co
 				: getReadContext();
```

Comment:
Should `getReadContext()` be renamed to `createReadContext()` to help avoid this kind of bug in the future?

Reply:
It doesn't necessarily always create one. In that method we check the Spring AOP support in there to see if there is an existing transaction context available to use and create the single-use if there isn't.  @meltsufin

Reply:
So why is the second call to `getReadContext()` not finding the transaction started by the previous call in this method?

Reply:
@meltsufin  Unlike Hibernate not every single operation happens inside explicit transactions that are opened and closed by the user or the framework .

When the user uses the transaction-anonymous-function support or the Transactional annotation then yes there is no problem before or now because we get the same transaction context.

however if the user was using this outside of any transaction boundaries (the  2 options mentioned above) then we use a single-use read context.